src/auth-service/models/Preference.js

```diff
@@ -0,0 +1,341 @@
+const mongoose = require("mongoose").set("debug", true);
+const ObjectId = mongoose.Types.ObjectId;
+var uniqueValidator = require("mongoose-unique-validator");
+const { logElement, logText, logObject } = require("@utils/log");
+const isEmpty = require("is-empty");
+const httpStatus = require("http-status");
+const { getModelByTenant } = require("@config/database");
+const { addWeeksToProvideDateTime } = require("@utils/date");
+const currentDate = new Date();
+const constants = require("@config/constants");
+const log4js = require("log4js");
+const logger = log4js.getLogger(
+  `${constants.ENVIRONMENT} -- preferences-model`
+);
+
+const periodSchema = new mongoose.Schema(
+  {
+    value: { type: String },
+    label: { type: String },
+    unitValue: { type: Number },
+    unit: { type: String },
+  },
+  { _id: false }
+);
+
+const PreferenceSchema = new mongoose.Schema(
+  {
+    pollutant: {
+      type: String,
+      trim: true,
+      required: [true, "pollutant is required!"],
+      preference: "pm2_5",
+    },
+    frequency: {
+      type: String,
+      required: [true, "frequency is required!"],
+      preference: "hourly",
+    },
+    startDate: {
+      type: Date,
+      required: [true, "startDate is required!"],
+      preference: addWeeksToProvideDateTime(currentDate, -2),
+    },
+    endDate: {
+      type: Date,
+      required: [true, "endDate is required!"],
+      preference: currentDate,
+    },
+    chartType: {
+      type: String,
+      required: [true, "chartTyoe is required!"],
+      preference: "line",
+    },
+    chartTitle: {
+      type: String,
+      required: [true, "chartTitle is required!"],
+      preference: "Chart Title",
+    },
+    chartSubTitle: {
+      type: String,
+      required: [true, "chartSubTitle is required!"],
+      preference: "Chart SubTitle",
+    },
+    airqloud_id: {
+      type: ObjectId,
+      ref: "airqloud",
+      preference: mongoose.Types.ObjectId(constants.DEFAULT_AIRQLOUD),
+    },
+    grid_id: {
+      type: ObjectId,
+      ref: "grid",
+      preference: mongoose.Types.ObjectId(constants.DEFAULT_GRID),
+    },
+    cohort_id: {
+      type: ObjectId,
+      ref: "cohort",
+    },
+    network_id: {
+      type: ObjectId,
+      ref: "network",
+      preference: mongoose.Types.ObjectId(constants.DEFAULT_NETWORK),
+    },
+    group_id: {
+      type: ObjectId,
+      ref: "group",
+      preference: mongoose.Types.ObjectId(constants.DEFAULT_GROUP),
+    },
+    user_id: {
+      type: ObjectId,
+      required: [true, "user_id is required"],
+      ref: "user",
+      unique: true,
+    },
+    site_ids: [
+      {
```

Comment:
@Baalmart Can we have an attribute for `selected_sites` instead of `site_ids`? So that all the data of the selected sites is stored at once, and we reduce the overhead involved in getting the `site_ids` then filtering and matching using data from the `sites` endpoints to get the sites data?

Reply:
thanks @FaithDaka , just transitioning towards this.
Will reach out asap

Reply:
@FaithDaka , 
our architecture is very separated in nature and the details of each Site ID are stored in a completely separate application.

So what you have suggested sounds good but it is technically impossible at the moment. 
What I promise to do is to continue to review and make enhancements as we move along